### DISCUSSION TITLE
Lesson4 : I can't print compiled_sol

### DISCUSSION THREAD

**Discussion by wilonweb** [smartcontractkit/full-blockchain-solidity-course-py#759] [Q&A] [answered]
Hello 

I can't get the object from solcx 
I don't know if is my code or if I had a problem with solcx. 

If i launch `python deploy.py` on terminal 

`Information : impossible de trouver des fichiers pour le(s) modèle(s) spécifié(s).
Traceback (most recent call last):
  File "D:\Programmations\smartContract\demo\web3_py-simple_storage\deploy.py", line 10, 
in <module>
    compiled_sol = compile_standard(
  File "C:\Users\wilfr\AppData\Roaming\Python\Python39\site-packages\solcx\main.py", line 371, in compile_standard
    solc_binary = get_executable(solc_version)
  File "C:\Users\wilfr\AppData\Roaming\Python\Python39\site-packages\solcx\install.py", line 194, in get_executable
    raise SolcNotInstalled(
solcx.exceptions.SolcNotInstalled: solc 0.6.0 has not been installed. Use solcx.install_solc('0.6.0') to install.
PS D:\Programmations\smartContract\demo\web3_py-simple_storage> `
 
 And at the bottom is my deploy.py file. 

 ```Python
from solcx import compile_standard

with open(
    "./Simplestorage.sol", "r"
) as file:  # On ouvre le ficheir simpleStorage, en mode read et nous l'apelon file
    simple_storage_file = file.read()  # On créer une variable pour lire le file

    # Compile our solidity

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {
                    "*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)
print(compiled_sol)

```
 

And if i launch `pip install py-solc-x` on the terminal i got 

`Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: py-solc-x in c:\users\wilfr\appdata\roaming\python\python39\site-packages (1.1.1)
Requirement already satisfied: requests<3,>=2.19.0 in c:\users\wilfr\appdata\roaming\python\python39\site-packages (from py-solc-x) (2.26.0)
Requirement already satisfied: semantic-version<3,>=2.8.1 in c:\users\wilfr\appdata\roaming\python\python39\site-packages (from py-solc-x) (2.8.5)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\users\wilfr\appdata\roaming\python\python39\site-packages (from requests<3,>=2.19.0->py-solc-x) (1.26.6)
Requirement already satisfied: charset-normalizer~=2.0.0 in c:\users\wilfr\appdata\roaming\python\python39\site-packages (from requests<3,>=2.19.0->py-solc-x) (2.0.3)
Requirement already satisfied: idna<4,>=2.5 in c:\users\wilfr\appdata\roaming\python\python39\site-packages (from requests<3,>=2.19.0->py-solc-x) (3.2)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\wilfr\appdata\roaming\python\python39\site-packages (from requests<3,>=2.19.0->py-solc-x) (2021.5.30)`

**Comment by cromewar** [comment, marked as answer]
Hello @NinjaTurtles-cloud you are missing this line:
`install_solc("0.6.0")`before `compiled_sol`

**Reply by wilonweb** [reply]
Thanks 
I had to change the first two line of code with that one.
```from solcx import compile_standard, install_solc  # <- import the install_solc method!
install_solc("0.6.0")  # <- Add this line and run it at least once!```

( the import import the install_solc method is hide .... )

**Reply by cromewar** [reply]
I'm glad is solved now
